src/visualization/components/View.test.tsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react'
+import {render} from '@testing-library/react'
+import {AnnotationsList, RemoteDataState} from 'src/types'
+
+import Band from 'src/visualization/types/Band/index'
+import Check from 'src/visualization/types/Check/index'
+import Gauge from 'src/visualization/types/Gauge/index'
+import Graph from 'src/visualization/types/Graph/index'
+import Heatmap from 'src/visualization/types/Heatmap/index'
+import Histogram from 'src/visualization/types/Histogram/index'
+import GeoMap from 'src/visualization/types/Map/index'
+import Mosaic from 'src/visualization/types/Mosaic/index'
+import Scatter from 'src/visualization/types/Scatter/index'
+import SimpleTable from 'src/visualization/types/SimpleTable/index'
+import SingleStat from 'src/visualization/types/SingleStat/index'
+import SingleStatWithLine from 'src/visualization/types/SingleStatWithLine/index'
+import Table from 'src/visualization/types/Table/index'
+
+const supportedVisualizations = {}
+
+Band(visualization => {
+  supportedVisualizations[visualization.type] = visualization
+})
+Check(visualization => {
+  supportedVisualizations[visualization.type] = visualization
+})
+Gauge(visualization => {
+  supportedVisualizations[visualization.type] = visualization
+})
+Graph(visualization => {
+  supportedVisualizations[visualization.type] = visualization
+})
+Heatmap(visualization => {
+  supportedVisualizations[visualization.type] = visualization
+})
+Histogram(visualization => {
+  supportedVisualizations[visualization.type] = visualization
+})
+GeoMap(visualization => {
+  supportedVisualizations[visualization.type] = visualization
+})
+Mosaic(visualization => {
+  supportedVisualizations[visualization.type] = visualization
+})
+Scatter(visualization => {
+  supportedVisualizations[visualization.type] = visualization
+})
+SimpleTable(visualization => {
+  supportedVisualizations[visualization.type] = visualization
+})
+SingleStat(visualization => {
+  supportedVisualizations[visualization.type] = visualization
+})
+SingleStatWithLine(visualization => {
+  supportedVisualizations[visualization.type] = visualization
+})
+Table(visualization => {
+  supportedVisualizations[visualization.type] = visualization
+})
+
+jest.doMock('src/visualization/contextLoader', () => {
+  return {
+    ...(jest as any).requireActual('src/visualization/contextLoader'),
+    buildSupportedVisualizations: jest.fn(() => {
+      return supportedVisualizations
+    }),
+  }
+})
```

Comment:
this replicates the functionality in `src/visualization/contextLoader`. This is why we'll need to update this test every time we add a new visualization - it doesn't automagically get added like the code under test.